src/outfit.ts

```diff
@@ -470,6 +470,39 @@ export class Outfit {
     return success ? outfit : null;
   }
 
+  /**
+   * Add a rider to the outfit.
+   *
+   * This function does *not* equip the corresponding item; it must be equipped separately.
+   *
+   * If a familiar is already specified as the rider that is different from the provided target, this function will return false and not change the rider.
+   * @param target The familiar to use as the rider, or a ranked list of familiars to try to use as the rider.
+   * @returns True if we successfully set the slot to a valid rider.
+   */
+  private equipRider(target: Familiar | Familiar[], slot: Slot): boolean {
+    const current = this.riders.get(slot);
+    const targets = Array.isArray(target) ? target : [target];
+
+    if (current) {
+      return targets.includes(current);
+    }
+
+    // Gather the set of riders that are equipped in other rider slots.
+    const other_riders: Familiar[] = [];
```

Comment:
is there a particular pattern here for when you choose to use snake_case over camelCase?

Reply:
also I think you can just do
```
const other_riders = [...this.riders.entries()].filter(([other_rider_slot]) => slot !== other_rider_slot).map(([, value]) => value)
```

Reply:
The pattern is that my day job is almost entirely python and sometimes that snake_case leaks into my typescript. Fixed.